### PR TITLE
update(devops): Add new content

### LINKED ISSUE #237: What is DevOps
DevOps is the practice & culture of applying lean manufacturing principles to the IT Value stream to increase software velocity, reduce waste, & improve quality with an emphasis on leveraging *people, processes, & tools.*

It focuses on unifying Development & IT Operations through the principles of flow, feedback, & experimentation.

[https://www.youtube.com/watch?v=OOGFxVzW35w](https://www.youtube.com/watch?v=OOGFxVzW35w)

# Key Definitions

- Continuous Integration (CI): Addresses building software from contributions of individual developers in a controlled, agile way. (Checking all code into centralized repo daily)
- Continuous Deployment (CD): Automated Deployment (or transport) of code to production with no manual steps required. This will not be the plan with all applications.
- Continuous Delivery (CD): Automation that ensures deployment is validated, however manual step still required to deploy to production.
- Deployment: The deployment (or transport) of code to a higher-level environment, up to & including production.
- Release: The defined scope & business objective for a set of features to be delivered in a certain timeline.

# Why do DevOps?

In 2012, Puppet Labs “state of DevOps report” benchmarked 4,039 IT orgs using DevOps and found:

- 30x more frequent code deployments
- 8,000 x faster code deployment lead time
- 2x the change success rate
- 12x faster MTTR (mean time to repair)

# Some examples of why do DevOps?

“*The competitive advantage this capability creates is enormous, enabling faster feature time to market, increased customer satisfaction, market share, employee productivity, and happiness.”*

**-Phoenix Project**

image

# Areas of Emphasis

DevOps principles are applied to and assisted by the following 3 Key areas all working together:

- People - DevOps relies on strong people & a strong culture of experimentation, learning, collaboration, and shared goals between Development, QA & IT Operations.
- Processes – DevOps builds further upon Lean Agile, applying concepts of lean manufacturing to IT value streams.
- Tools– DevOps relies on enhanced tools & technologies, including automation, telemetry, & collaboration tools.

# What Is the Goal of DevOps?

Improve collaboration between all stakeholders from planning through delivery and automation of the delivery process in order to:

- Improve deployment frequency
- Achieve faster time to market
- Lower failure rate of new releases
- Shorten lead time between fixes
- Improve mean time to recovery

# CI Systems - Examples

image

# CD Systems - Examples

image

# References

“The principles behind DevOps are the same that transformed manufacturing. Instead of optimizing how raw materials are transformed into finished goods, DevOps shows how we optimize the IT value stream..” - Phoenix Project

“Imagine living in a DevOps world, where product owners, Development, QA, IT Ops, and InfoSec work together relentlessly to help each other and the overall organization win. They are enabling fast flow of planned work into production, while preserving world-class stability, reliability, availability, and security.” - Phoenix Project

[Periodic Table of DevOps Tools](https://www.notion.so/Periodic-Table-of-DevOps-Tools-4e9b29562a944132b1235cfa7e400d68)

[Automation](https://www.notion.so/Automation-e4dee021a4494d9cbdebb3ebc6205f4f)

### PR DESCRIPTION
**Pull request by Edgar097**
### GH Issue

Resolves #237  

### Steps to test
1.  Open DevOps
1. Check Content

#### CheckList
- [ ] Follow proper Markdown format
- [ ] The content is adequate
- [ ] The content is available in both english and spanish
- [ ] I Ran a spell check

### CODE REVIEW

**Comment by Edgar097** [comment]
There are several images that I need to complete this section , however it will be nice if @ldrojas and @xavier506 can check the content and the distribution of the infromation. Thank you

**Review by xavier506 — DISMISSED** [review]
LGTM

**Review by steph222 — APPROVED** [review]

**Review by steph222 — APPROVED** [review]

File: docs/devops.md [hunk 1]
```diff
@@ -4,13 +4,66 @@ title: DevOps
...
+- 2x the change success rate
+- 12x faster MTTR (mean time to repair)
+
+### Some examples of why do DevOps?
+
+“*The competitive advantage this capability creates is enormous, enabling faster feature time to market, increased customer satisfaction, market share, employee productivity, and happiness.”*
+
+**-Phoenix Project**
+
+![https://s3-us-west-2.amazonaws.com/secure.notion-static.com/c7bc6509-de0a-4efd-aa9c-f3a4486809b5/Untitled.png](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/c7bc6509-de0a-4efd-aa9c-f3a4486809b5/Untitled.png)
```
Comment (leisterfrancisco) [inline docs/devops.md:39]: I can't see this content

File: docs/devops.md [hunk 2]
```diff
@@ -4,13 +4,66 @@ title: DevOps
...
+
+- Improve deployment frequency
+- Achieve faster time to market
+- Lower failure rate of new releases
+- Shorten lead time between fixes
+- Improve mean time to recovery
+
+## CI Systems - Examples
+
+![https://s3-us-west-2.amazonaws.com/secure.notion-static.com/70d28084-e01c-4807-9dce-d106519ba31c/CI_Example.png](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/70d28084-e01c-4807-9dce-d106519ba31c/CI_Example.png)
```
Comment (leisterfrancisco) [inline docs/devops.md:61]: I can't see this content

File: docs/devops.md [hunk 3]
```diff
@@ -4,13 +4,66 @@ title: DevOps
...
+
+## CD Systems - Examples
+
+![https://s3-us-west-2.amazonaws.com/secure.notion-static.com/db2bfb4e-779e-4c7b-9933-05cb788f5008/CD_Example.png](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/db2bfb4e-779e-4c7b-9933-05cb788f5008/CD_Example.png)
```
Comment (leisterfrancisco) [inline docs/devops.md:65]: I can't see this content

**Review by leisterfrancisco — CHANGES_REQUESTED** [review]

**Review by leisterfrancisco — DISMISSED** [review]
Good work 👍

File: docs/devops.md [hunk 4]
```diff
@@ -4,13 +4,66 @@ title: DevOps
...
 
-Continuous integration allows us to detect problems early reducing costs and help us deliver robust software more rapidly.
+<figure class="video_container">
+  <iframe width="100%" height="315" src="https://www.youtube.com/embed/OOGFxVzW35w" frameborder="0" allowfullscreen="true">
+  </iframe>
+</figure>
+
+## Key Definitions
+
+- Continuous Integration (CI): Addresses building software from contributions of individual developers in a controlled, agile way. (Checking all code into centralized repo daily)
```
Comment (xavier506) [inline docs/devops.md:18]: lets bold list item titles 

- **Continuous Integration (CI)**: Addresses bui

File: docs/devops.md [hunk 5]
```diff
@@ -95,29 +137,39 @@ https://www.youtube.com/embed/EnJ7qX9fkcU&list=PL7bmigfV0EqQt5_pBPQ8tsZjI1w68-e0
...
-https://computingforgeeks.com/how-to-install-minikube-on-ubuntu-debian-linux/. 
+[Install Minikube](https://computingforgeeks.com/how-to-install-minikube-on-ubuntu-debian-linux/) 
 
 Request to the administrators for access to our Openstack installation dashboard and the Sandbox
 project. Then you can follow the next steps:
 
 - First of all connect to the master node using the SSH tool:
 
-```
+```shell
```
Comment (xavier506) [inline docs/devops.md:152]: 👍

**Review by xavier506 — CHANGES_REQUESTED** [review]
please see minor formatting changes

**Review by xavier506 — APPROVED** [review]
LGTM

**Review by xavier506 — DISMISSED** [review]

**Review by xavier506 — APPROVED** [review]
LGTM

**Review by steph222 — APPROVED** [review]

File: docs/devops.md [hunk 6]
```diff
@@ -4,19 +4,76 @@ title: DevOps
...
+
+## Key Definitions
+
+- **Continuous Integration (CI)**: Addresses building software from contributions of individual developers in a controlled, agile way. (Checking all code into centralized repo daily)
+- **Continuous Deployment (CD)**: Automated Deployment (or transport) of code to production with no manual steps required. This will not be the plan with all applications.
+- **Continuous Delivery (CD)**: Automation that ensures deployment is validated, however manual step still required to deploy to production.
+- **Deployment**: The deployment (or transport) of code to a higher-level environment, up to & including production.
+- **Release**: The defined scope and business objective for a set of features to be delivered in a certain timeline.
+
+## Why do DevOps?
```
Comment (xavier506) [inline docs/devops.md:24]: Why implement DevOps?

File: docs/devops.md [hunk 7]
```diff
@@ -4,19 +4,76 @@ title: DevOps
...
+- 2x the change success rate
+- 12x faster MTTR (mean time to repair)
+
+***
+
+>“*The competitive advantage this capability creates is enormous, enabling faster feature time to market, increased customer satisfaction, market share, employee productivity, and happiness.”*
+
+***
+
+**-Phoenix Project**
```
Comment (xavier506) [inline docs/devops.md:39]: Let's add `**-Phoenix Project**` inside the quote block like this : 

***
>“*The competitive advantage this capability creates is enormous, enabling faster feature time to market, increased customer satisfaction, market share, employee productivity, and happiness.”*
**-Phoenix Project**
***

**Review by xavier506 — APPROVED** [review]
LGTM

**Review by leisterfrancisco — APPROVED** [review]
Good work 👍

**Review by steph222 — APPROVED** [review]
Good work!!

**Pull request merged**